packages/core/src/operator/operator-runtime/processNewChallenge.ts

```diff
@@ -0,0 +1,146 @@
+                });
+                if (foundSubmission) {
+                    hasSubmission = true;
+                }
+            } else {
+                const submission = await retry(() => getBulkSubmissionForChallenge(challengeId, ownerOrPool.address), 3);
+                if (submission.submitted) {
+                    hasSubmission = true;
+                }
+            }
+
+            if (hasSubmission) {
+                operatorState.cachedLogger(`Address ${ownerOrPool.address} has submitted for challenge ${challengeId} by another node. If multiple nodes are running, this message can be ignored.`);
+                updateSentryAddressStatus(ownerOrPool.address, NodeLicenseStatus.WAITING_FOR_NEXT_CHALLENGE);
+                continue;
+            }
+
+            try {
+                //Try to submit once, if we get error 54 we don't need to try again. Any other error, we should give it 2 more tries.
+                await submitBulkAssertion(ownerOrPool.address, challengeId, challenge.assertionStateRootOrConfirmData, operatorState.cachedSigner, operatorState.cachedLogger);
+            } catch (error: any) {
+                if (error && error.message && error.message.includes('execution reverted: "54"')) {
+                    // If error code is 54, we don't need to log the error, it just means for this pool we already submitted
+                    operatorState.cachedLogger(`Address ${ownerOrPool.address} has submitted for challenge ${challengeId} by another node. If multiple nodes are running, this message can be ignored.`);
+                    updateSentryAddressStatus(ownerOrPool.address, NodeLicenseStatus.WAITING_FOR_NEXT_CHALLENGE);
+                    continue;
+                }
+
+                await retry(() => submitBulkAssertion(ownerOrPool.address, challengeId, challenge.assertionStateRootOrConfirmData, operatorState.cachedSigner, operatorState.cachedLogger), 2);
+            }
+
+            operatorState.cachedLogger(`Successfully submitted assertion for ${ownerOrPool.address}`);
+            updateSentryAddressStatus(ownerOrPool.address, NodeLicenseStatus.WAITING_FOR_NEXT_CHALLENGE);
+
+        } catch (error: any) {
+            operatorState.cachedLogger(`Error submitting assertion for address ${ownerOrPool.address} to challenge ${challengeId} - ${error && error.message ? error.message : error}`);
+            continue;
+        }
+    }
+}
+
+/**
+ * Helper function to calculate the boost factor off-chain. This requires the current Referee config from the subgraph
+ * 
+ * @param stakedEsXAIAmount amount of staked esXAI
+ * @param keyCount amount of staked keys in a pool / amount of unstaked keys for an owner, used to calculate the maxStakeAmount
+ * @param {RefereeConfig} refereeConfig the Referee config from the subgraph
+ * @return {bigint} The payout chance boostFactor. 200 for double the chance.
+ */
+const getBoostFactor = (
```

Comment:
Should we rename to `calculateBoostFactor` or `calculateBoostFactorLocally` to help distinguish the source of the boost factor?

Any benefit to making this a util function that could be used in other places?

Reply:
Renamed to `calculateBoostFactorLocally`.
We had this for V1 but never used it outside the operatorRuntime as it requires the refereeConfig from the subgraph.
If we need to use it in the future Im fine with moving it into its own exported function.